ding_dibs.module

```diff
@@ -420,3 +442,142 @@ function ding_dibs_cards() {
+    $today_date = format_date(REQUEST_TIME, 'custom', 'd/m/Y/H:i');
+    $start_date = format_date($start_val, 'custom', 'd/m/Y/H:i');
+    if (!empty($start_operation)) {
+      $start_operation_date = format_date($start_operation, 'custom', 'd/m/Y');
+    }
+    if ($today_date == $start_date || $today_date >= $start_date) {
+      if (empty($start_operation_date) || $start_operation_date < format_date(REQUEST_TIME, 'custom', 'd/m/Y')) {
+        $start_cron = TRUE;
+      }
+    }
+  }
+  if ($start_cron) {
+    if (!empty($emails_list)) {
+      $emails_array = preg_split("/\,|\r\n|\n/", $emails_list);
+      $emails_count = count($emails_array);
+      if ($emails_count >= 1) {
+        module_load_include('inc', 'ding_dibs', 'ding_dibs.email_reports');
+        $text_message = ding_dibs_email_reports();
+        variable_set('ding_dibs_start_operation_cron', REQUEST_TIME);
+        $id = _ding_dibs_insert_table(variable_get('ding_dibs_start_operation_cron', REQUEST_TIME));
+        for ($count = 0; $count < $emails_count; $count++) {
+          $email = $emails_array[$count];
+          if (!empty($email)) {
+            drupal_mail('system', 'mail', $email, language_default(), array(
+              'context' => array(
+                'subject' => t('Recent unusual DIBS reports'),
+                'message' => $text_message,
+              ),
+            ));
+          }
+        }
+        if (!empty($id)) {
+          _ding_dibs_ubdate_table($id, $start, $start_minute);
+        }
+      }
+    }
+  }
+}
+
+/**
+ * Insert information in a table ding_dibs_report_cron.
+ *
+ * @return
+ *   Id outcome writing new values to the table ding_dibs_report_cron.
+ */
+function _ding_dibs_insert_table($start) {
+  $query = db_insert('ding_dibs_report_cron');
+  $query->fields(array('time_start' => $start));
+  $id = $query->execute();
+  return $id;
+}
+
+/**
+ * Updating information in a table ding_dibs_report_cron.
+ *
+ * @param $id
+ *   id database table ding_dibs_report_cron.
+ *
+ * @param $start
+ *   Hours run cron.
+ *
+ * @param $start_minute
+ *   Minutes run cron.
+ */
+function _ding_dibs_ubdate_table($id, $start, $start_minute) {
+  $previous = _ding_dibs_select_info($id, 1, TRUE);
+  if (empty($previous)) {
+    $last = _ding_dibs_select_info($id, 1);
+  }
+  $start_minute = str_pad($start_minute, 2, "0", STR_PAD_LEFT);
+  $day = format_date(REQUEST_TIME, 'custom', 'd');
+  $hour = format_date(variable_get('ding_dibs_start_operation_cron', REQUEST_TIME), 'custom', 'H:i');
+  if ($last) {
+    $count = count($last) - 1;
+    if (!empty($last[$count]->time_start)) {
+      $last_day = format_date($last[$count]->time_start, 'custom', 'd');
+    }
+  }
+  else {
+    $last_day = $day - 1;
+  }
+  $message = $day . 'th september ' . $hour . ': Job delivered report for '
+    . $last_day . 'th sep ' . $start . ':' . $start_minute . ' - ' . $day . 'th september '
+    . $start . ':' . $start_minute . '.';
```

Comment:
The message is created and saved in DB (but not formed) when the page loads. Is has been done, so that if cron have not been working for a few days (maybe due to a server problem), the large volumes of reports will create load on page report. Simple sample messages from the database is handles these large volumes.